tests/core/test_indexset.py

```diff
@@ -85,20 +83,20 @@ def test_add_elements(self, test_mp, request):
 
         test_elements_2 = ["One", 2, 3.141]
         indexset_5 = run.optimization.indexsets.create("Indexset 5")
-        indexset_5.add(test_elements_2)
+        indexset_5.add(test_elements_2)  # type: ignore
         assert indexset_5.elements == test_elements_2
 
-    def test_list_indexsets(self, test_mp, request):
-        test_mp: Platform = request.getfixturevalue(test_mp)  # type: ignore
-        run = test_mp.runs.create("Model", "Scenario")
+    def test_list_indexsets(self, platform: ixmp4.Platform):
+        run = platform.runs.create("Model", "Scenario")
+        # Per default, list() lists only `default` version runs:
+        run.set_as_default()
```

Comment:
This line and the above line were deleted on purpose in the earlier PR since they are not necessary. Maybe they should be, but in my view they should then be introduced again when the underlying data structures are changed and actually require them.

Reply:
Is this what you mean with?

> In general, I'm fine with migrating to fixture-based/csv data. Some of the changes proposed here seem to revert intentional changes we made before, though, so I won't approve just yet.

Reply:
Ok I removed all the setting-as-default from the optimisation tests! Thanks for the suggestions, you can make multiline-suggestions by selecting multiple lines and commenting a "suggestion" code block. btw: 
![image](https://github.com/user-attachments/assets/58091702-500a-4279-b99a-6be3484a834b)